### PR TITLE
Guestbook

### PR DESCRIPTION
**Pull request by dguendisch**
**How to categorize this PR?**
/area testing
/kind test
/priority normal

**What this PR does / why we need it**:
Azure guestbook tests are flaky, see #2423
This PR increases the timeout to wait for the guestbook app to serve its URLs from `10min` to `20min` and the timeout to wait for the namespace to be deleted (at the end of the test) from `5min` to `10min`.
It adds additional dumping logic to dump the nginx ingress controller logs at the end of a failed test so that we have a better chance to analyze the cause of the flakes.

**Which issue(s) this PR fixes**:
Maybe improves #2423, at least helps to better understand its potential causes.

### CODE REVIEW

**Review by schrodit — DISMISSED** [review]
lgtm

File: test/framework/dump.go [hunk 1]
```diff
@@ -209,6 +210,37 @@ func (f *GardenerFramework) dumpGardenerExtension(extension v1alpha1.Object) {
...
+		if err := f.DumpLogsForPodInNamespace(ctx, ctxIdentifier, k8sClient, namespace, pod.Name); err != nil {
+			result = multierror.Append(result, err)
+		}
+	}
+	return result
+}
+
+func (f *CommonFramework) DumpLogsForPodInNamespace(ctx context.Context, ctxIdentifier string, k8sClient kubernetes.Interface, podNamespace, podName string) error {
+	f.Logger.Infof("%s [NAMESPACE %s] [POD %s] [LOGS]", ctxIdentifier, podNamespace, podName)
+	request := k8sClient.Kubernetes().CoreV1().Pods(podNamespace).GetLogs(podName, &corev1.PodLogOptions{})
```
Comment (ialidzhikov) [inline test/framework/dump.go]: There is already `pkg/client/kubernetes.GetPodLogs` func. We can use it, right?

File: test/framework/dump.go [hunk 2]
```diff
@@ -209,6 +210,37 @@ func (f *GardenerFramework) dumpGardenerExtension(extension v1alpha1.Object) {
 	}
 }
 
+func (f *CommonFramework) DumpLogsForPodsWithLabelsInNamespace(ctx context.Context, ctxIdentifier string, k8sClient kubernetes.Interface, namespace string, labels client.MatchingLabels) error {
```
Comment (ialidzhikov) [inline test/framework/dump.go]: Nit: you could make it 1 idea more generic:

```suggestion
func (f *CommonFramework) DumpLogsForPodsWithLabelsInNamespace(ctx context.Context, ctxIdentifier string, k8sClient kubernetes.Interface, namespace string, opts ...client.ListOption) error {
```

File: test/framework/dump.go (hunk 1 above)
Comment (dguendisch) [reply test/framework/dump.go]: Right, thanks for pointing out.

File: test/framework/dump.go (hunk 2 above)
Comment (dguendisch) [reply test/framework/dump.go]: Good idea, thanks!

**Comment by dguendisch** [comment]
Addressed comments, PTAL

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**